### PR TITLE
Add tests for new remote architecture

### PR DESCRIPTION
**Pull request by eliecharra**

### CODE REVIEW

File: pkg/remote/scanner_test.go [hunk 1]
```diff
@@ -0,0 +1,138 @@
...
+	"github.com/cloudskiff/driftctl/pkg/resource"
+	resourceaws "github.com/cloudskiff/driftctl/pkg/resource/aws"
+
+	"github.com/cloudskiff/driftctl/pkg/terraform"
+	"github.com/cloudskiff/driftctl/test"
+	"github.com/cloudskiff/driftctl/test/goldenfile"
+	"github.com/stretchr/testify/assert"
+)
+
+func TestS3Bucket(t *testing.T) {
```
Comment (wbeuil) [inline pkg/remote/scanner_test.go:30]: All our former supplier tests will be there?

File: test/terraform/provider.go [hunk 2]
```diff
@@ -0,0 +1,20 @@
...
+
+import (
+	"os"
+
+	"github.com/cloudskiff/driftctl/pkg/output"
+	"github.com/cloudskiff/driftctl/pkg/remote/aws"
+	"github.com/cloudskiff/driftctl/pkg/terraform"
+)
+
+func InitTestAwsProvider(providerLibrary *terraform.ProviderLibrary, version string) (*aws.AWSTerraformProvider, error) {
```
[the remark this answers is not in the document]
Comment (eliecharra) [reply test/terraform/provider.go:11]: We can handle this when working on GH test imho

File: pkg/remote/scanner_test.go (hunk 1 above)
Comment (eliecharra) [reply pkg/remote/scanner_test.go:30]: We'll end with a very long test file yeah, any other suggestions ?

Comment (wbeuil) [reply pkg/remote/scanner_test.go:30]: Can we split scanner tests like repository tests, by service?

`s3_scanner_test.go`

Comment (eliecharra) [reply pkg/remote/scanner_test.go]: Should be better with 74fda03

Comment (wbeuil) [reply pkg/remote/scanner_test.go]: I'm good with that !

**Review by wbeuil — APPROVED** [review]

**Comment by eliecharra** [comment]
Fix #701

**Pull request merged**